api/include/opentelemetry/nostd/absl/README.md

```diff
@@ -0,0 +1,9 @@
+# Notes on Abseil Variant implementation
+
+This is a snapshot of Abseil Variant `absl::variant` from Abseil `v2020-03-03#8`.
```

Comment:
As this PR includes some sources files form abseil, is it compatible with our current license?

Reply:
Yes, it is the same license. Apache License v2.0. It is better than MPark in this regard. Removing MPark eventually, we can get rid of the Boost license requirement.